pkg/resources/streamlit_acceptance_test.go

```diff
@@ -0,0 +1,428 @@
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "show_output.0.owner_role_type", "ROLE"),
+
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.name", id.Name()),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.title", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.root_location", rootLocation),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.main_file", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.query_warehouse", warehouse.ID().Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.url_id"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.default_packages"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.user_packages.#"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.import_urls.#"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.external_access_integrations.#", "1"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.external_access_integrations.0", integrationId.Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.external_access_secrets"),
+				),
+			},
+			{
+				ConfigDirectory:   acc.ConfigurationDirectory("TestAcc_Streamlit/complete"),
+				ConfigVariables:   m(),
+				ResourceName:      "snowflake_streamlit.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAcc_Streamlit_Rename(t *testing.T) {
+	acc.TestAccPreCheck(t)
+	databaseId := acc.TestClient().Ids.DatabaseId()
+	schemaId := acc.TestClient().Ids.SchemaId()
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	newId := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	stage, stageCleanup := acc.TestClient().Stage.CreateStageInSchema(t, schemaId)
+	t.Cleanup(stageCleanup)
+	m := func(name string) map[string]config.Variable {
+		return map[string]config.Variable{
+			"database":  config.StringVariable(databaseId.Name()),
+			"schema":    config.StringVariable(schemaId.Name()),
+			"stage":     config.StringVariable(stage.ID().FullyQualifiedName()),
+			"name":      config.StringVariable(name),
+			"main_file": config.StringVariable("foo"),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.NetworkPolicy),
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Streamlit/basic"),
+				ConfigVariables: m(id.Name()),
```

Comment:
Could you add also a change of one param at the same time (e.g. comment) to show the rename is not influencing other alters.

Reply:
Done with `main_file`.